### PR TITLE
chore: add semantic-pull-request

### PR DESCRIPTION
**Pull request by kenji-miyake**
Depends on https://github.com/tier4/autoware-github-actions/pull/13.

### CODE REVIEW

File: .github/workflows/semantic-pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  semantic-pull-request:
+    uses: tier4/autoware-github-actions/.github/workflows/reusable-semantic-pull-request.yaml@main
```
Comment (kenji-miyake) [inline .github/workflows/semantic-pull-request.yaml:12]: Might be moved to autowarefoundation.

**Review by KeisukeShima — APPROVED** [review]

File: .github/workflows/semantic-pull-request.yaml (hunk 1 above)
Comment (kenji-miyake) [reply .github/workflows/semantic-pull-request.yaml:12]: https://github.com/autowarefoundation/autoware-github-actions/blob/e354248417106605f3a54d14717646347419a85d/.github/workflows/reusable-semantic-pull-request.yaml

Comment (kenji-miyake) [inline .github/workflows/semantic-pull-request.yaml:12]:
```suggestion
    uses: autowarefoundation/autoware-github-actions/semantic-pull-request@tier4/proposal
```

**Comment by kenji-miyake** [comment]
@KeisukeShima If you have time, could you set up `sync-files` action for this repository?

**Comment by kenji-miyake** [comment]
Added by #24.

**Pull request closed without merging**